pkg/resources/management.cattle.io/v3/fleetworkspace/validator.go

```diff
@@ -89,21 +91,18 @@ func (m *Mutator) Admit(request *admission.Request) (*admissionv1.AdmissionRespo
 	}
 	ns, err := m.namespaces.Create(&namespace)
 	if err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return nil, err
-		}
-		// check if user has enough privilege to create fleet admin rolebinding in the namespace
-		cr, err := m.clusterroles.Cache().Get(fleetAdminRole)
-		if err != nil {
-			return nil, err
-		}
-		response := &admissionv1.AdmissionResponse{}
-		auth.SetEscalationResponse(response, auth.ConfirmNoEscalation(request, cr.Rules, namespace.Name, m.resolver))
-
-		ns, err = m.namespaces.Get(namespace.Name, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
+		if errors.IsAlreadyExists(err) {
```

Comment:
This will make the mutator no longer [Idempotent](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#idempotence). kube-apiserver may call mutators multiple times for one request, which would cause this to fail the second time it is ran [more info on reinvocation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#reinvocation-policy)

One way to solve the original issue is to add labels on the created namespace that indicates it was created for fleet. Then you could deny the creation of a fleet workspace for any namespaces that do not have that label. And leave the current logic for a namespace that does have that label.

Reply:
This has been fix in https://github.com/rancher/webhook/pull/260/commits/803504b098c5917b1bdff52c8ecc7f91148d7df8. I also added unit tests.